src/Client/BuzzHouse/Generator/ServerSettings.cpp

```diff
@@ -338,18 +338,37 @@ std::unordered_map<String, CHSetting> serverSettings = {
      CHSetting(
          [](RandomGenerator & rg)
          {
-             const DB::Strings & choices
-                 = {"'default'",
-                    "'grace_hash'",
-                    "'direct, hash'",
-                    "'hash'",
-                    "'parallel_hash'",
-                    "'partial_merge'",
-                    "'direct'",
-                    "'auto'",
-                    "'full_sorting_merge'",
-                    "'prefer_partial_merge'"};
-             return rg.pickRandomlyFromVector(choices);
+             String res;
+             DB::Strings choices
+                 = {"auto",
+                    "default",
+                    "direct",
+                    "full_sorting_merge",
+                    "grace_hash",
+                    "hash",
+                    "parallel_hash",
+                    "partial_merge",
+                    "prefer_partial_merge"};
+
+             if (rg.nextBool())
+             {
+                 res = rg.pickRandomlyFromVector(choices);
+             }
+             else
+             {
+                 const uint32_t nalgo = (rg.nextMediumNumber() % static_cast<uint32_t>(choices.size())) + 1;
```

Comment:
Why the +1?

Reply:
The `%` operation will return a number between 0 and `number_algorithms - 1`. Without the `+1` the case where all the algorithms are listed in the SET command will never be generated.

Reply:
> Without the +1 the case where all the algorithms are listed in the SET command will never be generated

Can you elaborate on that? I still don't see how that +1 makes the SET to have all combinations.

Reply:
Without `+1`, the maximum possible number to be generated is 8, right? So, you get the range [0,8[ in the for loop, which doesn't cover all the elements in `choices`.

Reply:
Also in case of 0, the SET command would be empty, which is not useful.